src/proxy.js

```diff
@@ -26,12 +26,53 @@ module.exports = function proxyPolyfill() {
     return o ? (typeof o === 'object' || typeof o === 'function') : false;
   }
 
+  const $Object = Object;
+
+  let hasProto = true;
+  const objectCreate =
+    $Object.create ||
+    ((hasProto = !({ __proto__: null } instanceof $Object))
```

Comment:
This is hard to read or understand. It looks like you're seeing whether a random object, `{__proto__: null}`, is an `Object`. In Chrome this is `false`.

Is that right—is this an IE10 thing where it would be different? It also seems like this would always be the same result, can we cache it at the top of the file if important?

Reply:
This determines whether the current environment supports [`__proto__` Property Names in Object Initializers](https://tc39.es/ecma262/#sec-__proto__-property-names-in-object-initializers).